stabilize path operations, optimize remove_comments, add initial sample for documentation

test-fullautomation opened the pull request.

ThPoll committed:
stabilize path operations, optimize remove_comments, add initial sample for documentation

test-fullautomation commented:
Hi Son,
can you please review and test this changes with testsuites-management?
* I stabilized path functions. It was e.g. not possible to use relative paths when used by a python module (I created sample.py for this).
* I optimized the code for "remove_comments". Also /* */ multiline-comments are working now.
* I added a start for python samples.
Thank you,
 Thomas

namsonx commented:
Hello Thomas,

     I will test with testsuites-management package and inform you the result.

Thank you,
Son

test-fullautomation commented:
Hi Son,
thank you, but we need to talk before.
I assume that it will be not compatible to testsuites-management now. The atest is also not working anymore.
I have to explain why and we need to find the further strategy.
Thank you,
 Thomas

test-fullautomation commented on sample/sample.py:
```diff
@@ -0,0 +1,12 @@
+
+prepro=JsonPreprocessor.CJsonPreprocessor()
+
+data=prepro.jsonLoad(".\json\json_with_comment.json")
```
Hi Son,
important is that this use-case is working.
We assume that most of the users will not require this for robotframework, but for any  other python project.
Usually in this case the configuration data is either relative to the project, or refrerenced by means of an environment variable.
Thank you,
 Thomas

test-fullautomation commented on JsonPreprocessor/CJsonPreprocessor.py:
```diff
@@ -169,41 +223,22 @@ def __processImportFiles(self, input_data):
+        sContent=file.read()
+        file.close()
+
+        pattern = re.compile(r'//.*?$|/\*.*?\*/|\'(?:\\.|[^\\\'])*\'|"(?:\\.|[^\\"])*"', re.DOTALL | re.MULTILINE)
```
This allows /* */ multiline comments, too.

test-fullautomation commented on JsonPreprocessor/CJsonPreprocessor.py:
```diff
@@ -134,6 +136,50 @@ def __init__(self, syntax=CSyntaxType.json, currentCfg={}):
         self.currentCfg = currentCfg
         self.lUpdatedParams = []
             
+    #
```
Stable path functions taken from TML Framework.

ThPoll committed:
maintenance and improvements

- add annotations to functions
- absolute and relative paths are allowed for main file as well as [import] file.
- %envvariable% and {envvariable} syntax are allowed in paths
- [import] is now relative to the current file, instead of relative to the base file. This allows shorter paths and it allows to move whole sets of json files to other locations. Only the path to root file of a subset need to be changed in this case.
- all docstrings are now in rst syntax and maintained
- lot of documenation added
- f syntax is use for all exception strings.
- unified style of all exception strings.

ThPoll committed:
extend samples

ThPoll committed:
add licenze to sample files

test-fullautomation commented:
Must be tested and then merged together with
https://github.com/test-fullautomation/robotframework-testsuitesmanagement/pull/19
